style/sass/README.md

```diff
@@ -27,29 +27,178 @@
 * Use a `%` unit for the amount/weight when using Sass's color functions: `darken($color, 20%)`, not `darken($color, 20)`
 * Use a trailing comma after each item in a map, including the last item.
 
-## Order
-
-* Use alphabetical order for declarations.
-* Place `@extends` and `@includes` at the top of your declaration list.
-* Place media queries directly after the declaration list.
-* Place pseudo-classes and pseudo-elements second.
-* Place nested elements third.
-* Place nested classes fourth.
-
 ## Selectors
 
-* Don't use ID's for style.
-* Avoid over-qualified selectors: `h1.page-title`, `div > .page-title`
 * Use meaningful names: `$visual-grid-color` not `$color` or `$vslgrd-clr`.
-* Be consistent about naming conventions for classes. For instance, if a project is using BEM, continue using it, and if it's not, do not introduce it.
 * Use ID and class names that are as short as possible but as long as necessary.
 * Avoid nesting more than 3 selectors deep.
 * Avoid using comma delimited selectors.
 * Avoid nesting within a media query.
-* Don't concatenate selectors using Sass's parent selector (`&`).
 
 ## Organization
 
 * Use a `base` directory for styling element selectors, global variables, global extends and global mixins.
 * Use HTML structure for ordering of selectors. Don't just put styles at the bottom of the Sass file.
 * Avoid having files longer than 100 lines.
+
+---
+
+## Selectors
+
+### Selector specificity
+
+- Don't use ID selectors.
+- Avoid over-qualified selectors, e.g. `h1.page-title`.
+
+<details>
+
+#### Code examples
+
+`h1.page-title` carries a specificity of 2, but can be reduced to 1 by removing
+the `h1` type selector:
+
+```diff
+-h1.page-title
++.page-title {
```

Comment:
In other guideline repositories that I've seen, a lot of the code samples have comments like:

```rb
# bad
the_code_with_the_violation

# bad
other_code_with_a_violation

# good
code_without_the_violation
```

Is the idea here that the added portion of the diff is the "good" example, and the removed portion is the "bad" example?

It's worth noting that I'm opposed to labelling "good" and "bad". If you know of less judgmental labels, let's use those!

Reply:
Yeah, I've seen this a lot, too ([Dropbox](https://github.com/dropbox/css-style-guide) and [Airbnb](https://github.com/airbnb/swift) being some examples).

Your point about the labels “good” and “bad” are exactly why I didn't go down that path, but at the same time, I do like the clarity it may bring. 

> Is the idea here that the added portion of the diff is the "good" example, and the removed portion is the "bad" example? If so, have we explained that?

Yes, and no, that is not explained right now.

Definitely open to suggestions here…

Reply:
I also like this idea, and I think it would be especially helpful if the "bad" example was something we might see a lot or that could appear fine at first glace. What if the labels depended on the context of the example? For instance, in this one the labels could be `# too specific` and `# just specific enough` or `# harder to override` and `# specific but overrideable` or... I think you get my point 😅